examples/hosts/app-integration/schema-upgrade/src/migrator/migrator.ts

```diff
@@ -86,100 +108,133 @@ export class Migrator extends TypedEventEmitter<IMigratorEvents> implements IMig
         }
 
         const doTheMigration = async () => {
```

Comment:
General q (maybe not directly tied to this PR) - is user expected to write all this logic to handle their own use cases? Would it be helpful to export this as experimental "migration" package where user can consume simplified set of APIs, without needing to understand all the internal contraints and various concepts/DDS-es we rely on to achieve migration?

Reply:
I'm not sure what the long term goals are here. Maybe @ChumpChief has a better idea?

Reply:
I'll resolve the convo since it's not related to this PR, and we move it to general category questions.